app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,76 @@
+class QuestionsController < ApplicationController
+  before_action :find_question, only: [:show, :edit, :update, :destroy]
+  before_action :authenticate_user, only: [:new, :create, :edit, :update, :destroy]
+
+  def index
+    @question = Question.new
+    @questions = Question.order_by_recent
+  end
+
+  def trending
+    @questions = Question.order_by_trending
+    render :index
+  end
+
+  def votes
+    @questions = Question.order_by_votes
+    render :index
+  end
+
+  def search
+    @questions = Question.where("title ILIKE ?", "%#{params[:search]}%")
+    @questions.order_by_recent
+    render :index
+  end
+
+  def show
+    @comments = @question.comments
+    @answers = @question.answers.includes(:comments)
+    @vote = Vote.new
+    @answer = Answer.new
+    @comment = Comment.new
+  end
+
+  def new
+    @question = Question.new
+  end
+
+  def create
+    question = Question.new(question_params.merge(submitter: current_user))
+    if question.save
+      redirect_to question_path(question)
+    else
+      redirect_to new_question_path, flash: {error: "Invalid input: must include both title and content."}
```

Comment:
It might be preferable to either
a) re-render new so that the user doesn't necessarily lose the form input
b) Let rails display the errors (use question.errors)